enhancements/authentication/separate-oauth-resources.md

```diff
@@ -0,0 +1,142 @@
+---
+title: Separate-OAuth-API-Resources
+authors:
+  - "@deads2k"
+reviewers:
+  - "@sttts"
+approvers:
+  - "@derekwaynecarr"
+creation-date: 2019-10-16
+last-updated: 2019-10-16
+status: implementable
+see-also:
+replaces:
+superseded-by:
+---
+
+# Separate OAuth API-Resources
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Open Questions [optional]
+
+## Summary
+
+The API resources used by the OpenShift OAuth server will be moved to their own `openshift-oauth-apiserver` repository.
+The openshift-oauth-apiserver binary will be managed by the existing authentication-operator.
+
+## Motivation
+
+The OpenShift OAuth server requires several openshift API resources in order to function.
+Because the OAuth server enables access to the kube-apiserver, it needs to be able to run separately from the existing
+set of API resources that are focused on enabling a developer workflow.
+This narrowing of scope expands the deployment options for the OpenShift OAuth server.
+
+### Goals
+
+1. Make it possible to run the OpenShift OAuth server without installing all the developer focused API Resources.
+
+### Non-Goals
+
+1. Move routes out of the OpenShift API Server.
```

Comment:
seems like without Routes, which IIUC is a dependency for oauth, this component would be dependent on the openshift-apiserver as well which doesn't get us anything.  Is there a plan to remove this dependency on Routes so this could run without the openshift-apiserver?

Reply:
> seems like without Routes, which IIUC is a dependency for oauth, this component would be dependent on the openshift-apiserver as well which doesn't get us anything. Is there a plan to remove this dependency on Routes so this could run without the openshift-apiserver?

Routes look like an entirely separable component without any dependency on another openshift primitive. I suggest that if they want to move (and they probably should), they should move somewhere separate.

Also, the current route checks could be reconsidered as the true interface is public exposure at a known point and serving certificate.